src/runtime_src/hip/core/device.h

```diff
@@ -3,12 +3,51 @@
 #ifndef xrthip_device_h
 #define xrthip_device_h
 
+#include "core/common/api/handle.h"
+#include "xrt/xrt_device.h"
+
+#include <vector>
+
 namespace xrt::core::hip {
+/**
+ * typedef device_handle - opaque device handle
+ */
+typedef uint32_t device_handle;
+
+// forward declaration
+class context;
 
 class device
 {
+  xrt::device m_xrt_device;
+  uint32_t m_device_id = UINT32_MAX;
+  unsigned int m_flags;
+
+public:
+  explicit
+  device(uint32_t _device_id);
```

Comment:
No leaning or trailing underscore, please.
```suggestion
  device(uint32_t device_id);
```